spec/oas_v1.yaml

```diff
@@ -0,0 +1,497 @@
+          type: string
+        description:
+          summary: Description of the voting measure
+          description: Extended information and context around the proposal.
+          type: string
+        ordinal:
+          summary: Number of this proposal's order
+          description: > 
+            Numeric descriptor of this proposal's ordering among others; Calculated as the concatentation
+            of the proposal's block number left-padded by 10 0's, transaction index left-padded by 3 0's, 
+            and log index left-padded by 3 0's.
+          type: string
+        createBlock:
+          summary: Proposal creation block number
+          description: Block number on which this proposal was created.
+          type: string
+        startBlock:
+          summary: Starting block number
+          description: Block number at which this proposal is open for voting.
+          type: string
+        endBlock:
+          summary: Ending block number
+          description: Block number at which the proposal has closed for voting.
+          type: string
+        cancelledBlock: 
+          summary: Cancelled block number 
+          description: Block number at which this proposal has been canceled. Zero if not canceled.
+          type: string
+        executedBlock:
+          summary: Execution block number 
+          description: Block number at which this proposal has been executed. Zero if not executed.
+          type: string
+        proposalData:
+          $ref: "#/components/schemas/ExecutionData"
+        proposalType:
+          $ref: "#/components/schemas/BaseProposalType"
+    BaseProposalType:  
+      # ADD standard, advanced, optimistic, snapshot enum
+      summary: Metadata about the proposal
+      description: An object describing the particular voting parameters of a proposal.
+      type: object 
+      properties:
+        # How are optimistic and approval types modeled? is there separate logic other than the 
+        # config of the following parameters?
+        name: 
+          summary: Name of proposal type
+          description: Name of proposal type
+          type: string
+        proposalTypeId:
+          summary: Numeric id of the proposal type
+          description: Numeric id of the proposal type
+          type: integer
+        # Note: quorum and threshold are strings since I'm not sure about the unit/size here 
+        # (i.e. if they can be contained in an int32/int64 type)
+        quorum:
+          summary: Minimmum participation for passage
+          description: >
+            The minimum number of voting power needed to be involved in a given proposal as a prerequisite for passage
+          type: string 
+        approvalThreshold:
+          summary: Amount needed for measure to pass 
+          description: The amount of voting power needed to pass the given proposal.
+          type: string
+    ExtendedProposalType:
```

Comment:
Not sure we need to keep these separate, or include at all?

Reply:
Yep on second look, totally agree. would opt to include in base proposal type. in general, I think we might use the pattern of base/extended if expanding certain fields would be very expensive if we're doing a GET from a plural/non-scoped endpoint